components/admin/requests/forms/ReasonForReplacementForm.tsx

```diff
@@ -0,0 +1,217 @@
+          onChange={value =>
+            onChange({
+              ...reasonForReplacement,
+              reason: value as ReasonForReplacementEnum,
+            })
+          }
+        >
+          <Stack>
+            <Radio value={ReasonForReplacementEnum.Lost}>{'Lost'}</Radio>
+            <Radio value={ReasonForReplacementEnum.Stolen}>{'Stolen'}</Radio>
+            <Radio value={ReasonForReplacementEnum.Other}>{'Other'}</Radio>
+          </Stack>
+        </RadioGroup>
+      </FormControl>
+
+      {/* Conditionally render this section if Lost is selected as replacement reason */}
+      {reasonForReplacement.reason === ReasonForReplacementEnum.Lost && (
+        <>
+          <FormControl isRequired paddingBottom="24px">
+            <FormLabel>{`Date`}</FormLabel>
+            <Input
+              type="date"
+              value={
+                reasonForReplacement.lostTimestamp
+                  ? formatDate(new Date(reasonForReplacement.lostTimestamp), true)
+                  : ''
+              }
+              onChange={event => {
+                const updatedlostTimestamp = new Date(reasonForReplacement.lostTimestamp);
+                updatedlostTimestamp.setFullYear(parseInt(event.target.value.substring(0, 4)));
+                updatedlostTimestamp.setMonth(parseInt(event.target.value.substring(5, 7)) - 1);
+                updatedlostTimestamp.setDate(parseInt(event.target.value.substring(8, 10)));
+
+                onChange({
+                  ...reasonForReplacement,
+                  lostTimestamp: Date.parse(updatedlostTimestamp.toString())
+                    ? updatedlostTimestamp
+                    : reasonForReplacement.lostTimestamp,
+                });
+              }}
+            />
+          </FormControl>
+
+          <FormControl paddingBottom="24px">
+            <FormLabel>
+              {'Timestamp '}
+              <Box as="span" textStyle="body-regular">
+                {'(optional)'}
+              </Box>
+            </FormLabel>
+            <Input
+              placeholder={'eg. 04:00 pm'}
+              type="time"
+              value={
+                new Date(reasonForReplacement.lostTimestamp).toLocaleTimeString('en-US', {
+                  hour12: false,
+                  hour: '2-digit',
+                  minute: '2-digit',
+                }) || ''
+              }
+              onChange={event => {
+                const updatedlostTimestamp = new Date(reasonForReplacement.lostTimestamp);
+                updatedlostTimestamp.setHours(parseInt(event.target.value.substring(0, 2)));
+                updatedlostTimestamp.setMinutes(parseInt(event.target.value.substring(3, 5)));
+
+                onChange({
+                  ...reasonForReplacement,
+                  lostTimestamp: Date.parse(updatedlostTimestamp.toString())
+                    ? updatedlostTimestamp
+                    : reasonForReplacement.lostTimestamp,
+                });
+              }}
```

Comment:
Parses the time string inputted in the form field for the hour, and minute; then updates the timestamp with each while keeping the year, month, and day the same.